File: abc/abc377/b/main.go
package main

import (
    "bufio"
    "fmt"
    "os"
)

func main() {
    reader := bufio.NewReader(os.Stdin)
    
    var rowMask, colMask uint8
    
    board := make([]string, 8)
    
    for i := 0; i < 8; i++ {
        line, _ := reader.ReadString('\n')
        board[i] = line[:8]
        
        for j := 0; j < 8; j++ {
            if line[j] == '#' {
                rowMask |= 1 << i
                colMask |= 1 << j
            }
        }
    }
    
    safeSpaces := 0
    emptyRows := ^rowMask & 0xFF
    emptyCols := ^colMask & 0xFF
    
    for i := 0; i < 8; i++ {
        if emptyRows&(1<<i) != 0 {
            for j := 0; j < 8; j++ {
                if emptyCols&(1<<j) != 0 && board[i][j] == '.' {
                    safeSpaces++
                }
            }
        }
    }
    
    fmt.Println(safeSpaces)
}

File: abc/abc377/b/main.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let reader = stdin.lock();
    let mut row_mask: u8 = 0;
    let mut col_mask: u8 = 0;
    
    for (i, line) in reader.lines().take(8).enumerate() {
        let row = line.unwrap();
        
        for (j, ch) in row.chars().take(8).enumerate() {
            if ch == '#' {
                row_mask |= 1 << i;
                col_mask |= 1 << j;
            }
        }
    }

    let safe_rows = !row_mask;
    let safe_cols = !col_mask;
    let mut count = 0;
    
    for i in 0..8 {
        if safe_rows & (1 << i) != 0 {
            count += (safe_cols as u32).count_ones() as i32;
        }
    }
    
    println!("{}", count);
}